src/WebJobs.Script/Host/ScriptHostIdProvider.cs

```diff
@@ -65,7 +65,7 @@ internal static HostIdResult GetDefaultHostId(IEnvironment environment, ScriptAp
             {
                 // The hostid is derived from the hostname for Linux consumption.
                 string hostName = environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteHostName);
-                hostId = hostName?.Replace(".azurewebsites.net", string.Empty);
+                hostId = hostName?.Replace(environment.GetDnsSuffixWithDotPrefix(), string.Empty);
```

Comment:
Is it because the value we need to extract may itself contain periods that we can't take a parsing approach here (e.g. everything up to the first period) rather than a list of cloud suffixes that we'll have to ensure stays up to date over time?

Reply:
Also, once an app has ran and established/persisted an ID value, changing it later is very problematic. Won't these changes result in a changing ID for some apps at this point?

Reply:
Yes, would need some discussion on how to roll out this change to reduce breaks. Current options are appsetting to opt in to this behavior or make this change part of vnext

Regarding SC, it seems like the logic there is already out of sync with what we do here (both before and after this PR)
So I assume it is already broken there and we dont have to worry about that getting out of sync with host?

Reply:
@chiangvincent  - please open tracking item to keep SC logic that looks at HostId in sync with the host

> One more point - there's code in the ScaleController (GetHostIdFromSiteName) that computes the ID and that logic needs to be kept in sync with this logic. That logic doesn't do any of this suffix parsing - it's just using the site name.

Reply:
@balag0 / @mathewc - thinking more about this - we have another option 

- Provide opt-in flag for existing Cx
- Default the flag to true for new applications only through control plane changes in Antares

This will help make the change now without breaking existing apps but also will help address the broken scenarios without waiting for vNext. 

thoughts?

Reply:
Another option, and perhaps the safest, would be a fix in the storage layer to deal with the host IDs that have invalid names. Even the second option about having the flag defaulted to true could impact customer flows where they delete and recreate resources through automation.

Reply:
Yes you are right - recreated apps option 2 would not work. So, Opt-in flag for now and long term updates to storage layer?